examples/cve_published_last_month_export_xlsx.py

```diff
@@ -0,0 +1,117 @@
+"""Export active vulnerabilities published last month"""
+
+import os
+import datetime
+from configparser import ConfigParser
+import xlsxwriter
+from cbw_api_toolbox.cbw_api import CBWApi
+
+CONF = ConfigParser()
+CONF.read(os.path.join(os.path.abspath(os.path.dirname(__file__)), '..', 'api.conf'))
+CLIENT = CBWApi(CONF.get('cyberwatch', 'url'), CONF.get('cyberwatch', 'api_key'), CONF.get('cyberwatch', 'secret_key'))
+
+CLIENT.ping()
+
+def get_cyberwatch_cves(start_date, end_date):
+    """Returns a list of CVEs published between two dates"""
+    parameters = {"active": "true"}
+    exported_cves = []
+    cves = CLIENT.cve_announcements(parameters)
+
+    for cve in cves:
+        if(cve.published is not None and parse_cyberwatch_date(cve.published) >= start_date
+           and parse_cyberwatch_date(cve.published) <= end_date):
+            exported_cves.append(cve)
+
+    print("{} unique CVEs affecting your computers were published between {} and {}."
+          .format(len(exported_cves), start_date, end_date))
+    return exported_cves
+
+def parse_cyberwatch_date(cbwdate):
+    """Convert a date retrieved from Cyberwatch API to a datetime object"""
+    return datetime.datetime.strptime(cbwdate.split('T')[0], "%Y-%m-%d").date()
+
+def get_updates(cve_code, server):
+    """Retrieves affected technology for a given CVE on a server object"""
+    affected_technology = ""
+    affected_version = ""
+
+    for update in server["server"].updates:
+        if cve_code in update["cve_announcements"]:
+            if update["target"] is not None and update["target"]["product"] is not None:
+                affected_technology = update["target"]["product"]
+            if update["target"] is not None and update["target"]["version"] is not None:
+                affected_version = update["target"]["version"]
+        break
+    return affected_technology, affected_version
+
+def export_xls(cve_list, filename):
+    """Export a list of CVEs to an XLS file"""
+    xls_export = xlsxwriter.Workbook(filename)
+    row_unique_cve = 0
+    row_computer_cve = 0
+
+    tab_unique_cve = xls_export.add_worksheet("Vulnerabilities by CVE code")
+
+    tab_unique_cve.write(row_unique_cve, 0, "Vulnerabilty")
+    tab_unique_cve.write(row_unique_cve, 1, "CVSS score v3")
+    tab_unique_cve.write(row_unique_cve, 2, "CVSS score v2")
+    tab_unique_cve.write(row_unique_cve, 3, "Exploitable")
+    tab_unique_cve.write(row_unique_cve, 4, "Number of affected computers")
+    tab_unique_cve.write(row_unique_cve, 5, "Number of fixed computers")
+    tab_unique_cve.write(row_unique_cve, 6, "Publication date")
+
+    tab_computer_cve = xls_export.add_worksheet("Vulnerabilities by computer")
+
+    tab_computer_cve.write(row_computer_cve, 0, "Vulnerabilty")
```

Comment:
"Vulnerability"

Reply:
The error is related to the use of the exploitable attribute, it would be better to use exploit_code_maturity anyway.